kratos/geometries/nurbs_volume_geometry.h

```diff
@@ -0,0 +1,1112 @@
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Conctructor for B-Spline volumes
+    NurbsVolumeGeometry(
+        const PointsArrayType& rThisPoints,
+        const SizeType PolynomialDegreeU,
+        const SizeType PolynomialDegreeV,
+        const SizeType PolynomialDegreeW,
+        const Vector& rKnotsU,
+        const Vector& rKnotsV,
+        const Vector& rKnotsW)
+        : BaseType(rThisPoints, &msGeometryData)
+        , mPolynomialDegreeU(PolynomialDegreeU)
+        , mPolynomialDegreeV(PolynomialDegreeV)
+        , mPolynomialDegreeW(PolynomialDegreeW)
+        , mKnotsU(rKnotsU)
+        , mKnotsV(rKnotsV)
+        , mKnotsW(rKnotsW)
+    {
+        CheckAndFitKnotVectors();
+    }
+
+    /// Attention: Weigths are not yet implemented!
+    /// Conctructor for NURBS volumes
+    // NurbsVolumeGeometry(
+    //     const PointsArrayType& rThisPoints,
+    //     const SizeType PolynomialDegreeU,
+    //     const SizeType PolynomialDegreeV,
+    //     const SizeType PolynomialDegreeW,
+    //     const Vector& rKnotsU,
+    //     const Vector& rKnotsV,
+    //     const Vector& rKnotsW,
+    //     const Vector& rWeights)
+    //     : BaseType(rThisPoints, &msGeometryData)
+    //     , mPolynomialDegreeU(PolynomialDegreeU)
+    //     , mPolynomialDegreeV(PolynomialDegreeV)
+    //     , mPolynomialDegreeW(PolynomialDegreeW)
+    //     , mKnotsU(rKnotsU)
+    //     , mKnotsV(rKnotsV)
+    //     , mKnotsW(rKnotsW)
+    //     , mWeights(rWeights)
+    // {
+    //     CheckAndFitKnotVectors();
+
+    //     KRATOS_ERROR_IF(rWeights.size() != rThisPoints.size())
+    //         << "Number of control points and weights do not match!" << std::endl;
+    // }
+
+    explicit NurbsVolumeGeometry(const PointsArrayType& ThisPoints)
+        : BaseType(ThisPoints, &msGeometryData)
+    {
+    }
+
+    /// Copy constructor.
+    NurbsVolumeGeometry(NurbsVolumeGeometry<TContainerPointType> const& rOther)
+        : BaseType(rOther)
+        , mPolynomialDegreeU(rOther.mPolynomialDegreeU)
+        , mPolynomialDegreeV(rOther.mPolynomialDegreeV)
+        , mPolynomialDegreeW(rOther.mPolynomialDegreeW)
+        , mKnotsU(rOther.mKnotsU)
+        , mKnotsV(rOther.mKnotsV)
+        , mKnotsW(rOther.mKnotsW)
+    {
+    }
+
+    /// Copy constructor from a geometry with different point type.
+    template<class TOtherContainerPointType> NurbsVolumeGeometry(
+        NurbsVolumeGeometry<TOtherContainerPointType> const& rOther)
+        : BaseType(rOther, &msGeometryData)
+        , mPolynomialDegreeU(rOther.mPolynomialDegreeU)
+        , mPolynomialDegreeV(rOther.mPolynomialDegreeV)
+        , mPolynomialDegreeW(rOther.mPolynomialDegreeW)
+        , mKnotsU(rOther.mKnotsU)
+        , mKnotsV(rOther.mKnotsV)
+        , mKnotsW(rOther.mKnotsW)
+    {
+    }
+
+    /// Destructor.
+    ~NurbsVolumeGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
```

Comment:
You can comment this within one line. In the other geometries this is updated in a bit merged pr. This removed unnecessary copied comments.

/// Assignment operator.

Reply:
Sry, did not really understand. And what if I have info, e.g. @details .... @brief...